### PR TITLE
[SPARK-41321][CONNECT] Support target field for UnresolvedStar

### PR DESCRIPTION
**Pull request by dengziming**
### What changes were proposed in this pull request?
1. Support target field UnresolvedStar
2. UnresolvedStar can be used simultaneously with other expression.

### Why are the changes needed?
This is a necessary feature for UnresolvedStar

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added 2 new unit tests.

### CODE REVIEW

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -360,13 +360,8 @@ class SparkConnectPlanner(session: SparkSession) {
...
       logical.OneRowRelation()
     }
-    // TODO: support the target field for *.
     val projection =
-      if (rel.getExpressionsCount == 1 && rel.getExpressions(0).hasUnresolvedStar) {
-        Seq(UnresolvedStar(Option.empty))
-      } else {
-        rel.getExpressionsList.asScala.map(transformExpression).map(UnresolvedAlias(_))
-      }
+      rel.getExpressionsList.asScala.map(transformExpression).map(UnresolvedAlias(_))
```
Comment (amaliujia) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:364]: Though the code was written like this, I am seeing a problem that if the Expression is already a `Alias`, we should not wrap it into `UnresolvedAlias` again? 

I think you can write a test case like `SELECT 1 as col` then you will find the output schema is not `col` but a auto-generated alias.

**Comment by amaliujia** [comment]
cc @cloud-fan

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala (hunk 1 above)
Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:364]: I am actually not sure if Catalyst will smartly see there is a wrapped `Alias` thus ignore the `UnresolvedAlias`. At least it is worth to verify this case.

Comment (cloud-fan) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:364]: I think it will ignore. Can we add a test?

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:364]: +1 if there is no such test then add one.

Comment (dengziming) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:396]: Thank you for your review @cloud-fan @amaliujia , I have added one test case for it, PTAL.

**Review by cloud-fan — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]
lgtm

**Review by amaliujia — APPROVED** [review]
LGTM

**Pull request closed without merging**